pkg/autoscaler/autoscaler.go

```diff
@@ -146,22 +153,23 @@ type perPodAggregation struct {
 	accumulatedConcurrency float64
 	probeCount             int32
 	window                 time.Duration
-	lameduckTime           *time.Time
+	latestStatTime         *time.Time
+	isActivator            bool
 }
 
 // Aggregates the given concurrency
-func (agg *perPodAggregation) aggregate(concurrency float64) {
-	agg.accumulatedConcurrency += concurrency
+func (agg *perPodAggregation) aggregate(stat Stat) {
+	agg.accumulatedConcurrency += stat.AverageConcurrentRequests
 	agg.probeCount++
+	agg.setLatestStatTime(stat.Time)
 }
 
-// Registers the earliest lameduck metric received.
-func (agg *perPodAggregation) lameduck(t *time.Time) {
-	if agg.lameduckTime == nil {
-		agg.lameduckTime = t
-	}
-	if agg.lameduckTime.After(*t) {
-		agg.lameduckTime = t
+// Registers the latest metric received.
+func (agg *perPodAggregation) setLatestStatTime(t *time.Time) {
+	if agg.latestStatTime == nil {
+		agg.latestStatTime = t
+	} else if agg.latestStatTime.Before(*t) {
```

Comment:
you can fold these two cases, like:

```golang
if agg.latestStatTime == nil || agg.latestStatTime.Before(*t) {
    agg.latestStatTime = t
}
```

Wondering if its then easier just to put it into `aggregate`, but that's a nit.

Reply:
Moved to `aggregate`.